crates/biome_rowan/src/ast/batch.rs

```diff
@@ -509,9 +502,22 @@ where
                             }
 
                             let old = &root_string[range_start..range_end];
-                            let new = &optional_inserted_text.map_or(String::new(), |t| t);
 
-                            text_edit_builder.with_unicode_words_diff(old, new);
+                            match optional_inserted_text {
+                                None => {
+                                    text_edit_builder.with_unicode_words_diff(old, "");
+                                }
+                                Some(element) => match element {
+                                    SyntaxElement::Node(node) => {
+                                        text_edit_builder
+                                            .with_unicode_words_diff(old, &node.to_string());
+                                    }
```

Comment:
This is where we're diffing a node, and unfortunately we need to extract the entire string: tokens, trivia **and child nodes/tokens**